rest/rest-tests.ts

```diff
@@ -117,24 +118,25 @@ var promiseOrResponse = interceptor({
 });
 
 client = rest
-    .wrap<any>(defaultRequest)
-    .wrap<any>(hateoas)
-    .wrap<any>(location)
-    .wrap<any>(mime)
-    .wrap<any>(pathPrefix)
-    .wrap<any>(basicAuth)
-    .wrap<any>(oAuth)
-    .wrap<any>(csrf)
-    .wrap<any>(errorCode)
-    .wrap<any>(retry)
-    .wrap<any>(timeout)
-    .wrap<any>(jsonp)
-    .wrap<any>(xdomain)
-    .wrap<any>(xhr)
-    .wrap<any>(noop)
-    .wrap<any>(fail)
-    .wrap<any>(knownConfig, { prop: 'value' })
-    .wrap<any>(transformedConfig, { prop: 'value' });
+    .wrap(defaultRequest)
+    .wrap(hateoas)
+    .wrap(location)
+    .wrap(mime)
+    .wrap(pathPrefix)
+    .wrap(basicAuth)
+    .wrap(oAuth)
+    .wrap(csrf)
+    .wrap(errorCode)
+    .wrap(retry)
+    .wrap(template, { template: 'auth={token}', params: { token: 'hunter2' } })
+    .wrap(timeout)
+    .wrap(jsonp)
+    .wrap(xdomain)
+    .wrap(xhr)
+    .wrap(noop)
+    .wrap(fail)
+    .wrap<KnownConfig>(knownConfig, { prop: 'value' })
```

Comment:
Ahh, it's failing to derive the correct type for `knownConfig` as it doesn't have an `init` method in the config it passes to `interceptor`. There's probably some way to change the definition of `rest/interceptor` to allow it to correctly derive it.